Source/WebCore/rendering/style/RenderStyleInlines.h

```diff
@@ -936,17 +936,15 @@ inline TypographicMode RenderStyle::typographicMode() const
 
 inline bool isSkippedContentRoot(const RenderStyle& style, const Element* element)
 {
-    switch (style.contentVisibility()) {
-    case ContentVisibility::Visible:
+    if (style.contentVisibility() == ContentVisibility::Visible)
         return false;
-    case ContentVisibility::Hidden:
+    // FIXME: check more display types.
+    if (style.isDisplayTableOrTablePart() && style.display() != DisplayType::TableCaption)
+        return false;
```

Comment:
It's unfortunate that this is essentially a duplicate of shouldApplySizeOrStyleContainment . It would be good to link these 2 somehow, at a minimum with a comment in both places.